ui/apps/platform/src/Containers/Risk/RiskPageHeader.js

```diff
@@ -3,39 +3,40 @@ import PropTypes from 'prop-types';
 
 import entityTypes, { searchCategories } from 'constants/entityTypes';
 import PageHeader from 'Components/PageHeader';
-import URLSearchInput from 'Components/URLSearchInput';
 import {
     ORCHESTRATOR_COMPONENT_KEY,
     orchestratorComponentOption,
 } from 'Containers/Navigation/OrchestratorComponentsToggle';
+import SearchFilterInput from 'Components/SearchFilterInput';
+import useURLSearch from 'hooks/useURLSearch';
+import searchOptionsToQuery from 'services/searchOptionsToQuery';
 import CreatePolicyFromSearch from './CreatePolicyFromSearch';
 
-function RiskPageHeader({ autoFocusSearchInput, isViewFiltered, searchOptions }) {
+function RiskPageHeader({ isViewFiltered, searchOptions }) {
+    const { searchFilter, setSearchFilter } = useURLSearch();
     const subHeader = isViewFiltered ? 'Filtered view' : 'Default view';
-    const autoCompleteCategories = [searchCategories[entityTypes.DEPLOYMENT]];
+    const autoCompleteCategory = searchCategories[entityTypes.DEPLOYMENT];
 
-    let prependAutocompleteQuery;
     const orchestratorComponentShowState = localStorage.getItem(ORCHESTRATOR_COMPONENT_KEY);
-    if (orchestratorComponentShowState !== 'true') {
-        prependAutocompleteQuery = orchestratorComponentOption;
-    }
+    const prependAutocompleteQuery =
+        orchestratorComponentShowState !== 'true' ? orchestratorComponentOption : [];
     return (
         <PageHeader header="Risk" subHeader={subHeader}>
-            <URLSearchInput
+            <SearchFilterInput
                 className="w-full"
-                categoryOptions={searchOptions}
-                categories={autoCompleteCategories}
+                searchFilter={searchFilter}
+                searchOptions={searchOptions}
+                searchCategory={autoCompleteCategory}
                 placeholder="Add one or more resource filters"
-                autoFocus={autoFocusSearchInput}
```

Comment:
Does the search input no longer auto-focus, or is it handled differently?

Reply:
Ah I meant to comment on this. The focus behavior is identical without this line, but I wasn't totally sure why.

Visiting the page directly auto-focuses the search input (e.g. from the left nav), while visiting the page with a deployment ID select does not auto-focus the search (e.g. from one of the Risk links on the home page). Similarly, visiting the page from the left nav and then selecting an item in the table de-selects the search input.

I'll add some notes to the testing section of this PR.

Reply:
Same with the "Create Policy" action - I'll add notes in the test section.

Reply:
Actually now that I am trying to get screenshots for the focus behavior, it is working differently intermittently... I'll look into this more.

Reply:
Ok the intermittent behavior is due to a couple of things, and depends on the GraphQL cache. Internally the new search is set to _always_ autofocus, but it will also have `isDisabled` set to `true` if the search options are not present in the cache (negating the autofocus behavior).

I think the cleanest path forward to retain the exact autofocus behavior is to:
- Add passthrough props on the new search component to let us configure `autoFocus` instead of defaulting to `true`
- Remove the default value of `{searchOptions.length === 0}` for `isDisabled` in the new search component
- Control the `isDisabled` and `autoFocus` props at the call site for each instance of this component

This would let us override the behavior, but still makes the two props mutually exclusive. It looks like the old behavior is to not disable the component at all, even if the search option data has not loaded yet.

Not hard to implement, but would be a slight change to any part of the app using `<SearchFilterInput>`.

Reply:
By "old" behavior, I think you mean the <URLSearchInput> component. 

We don't have to maintain "backwards-compatibility" with that

Reply:
Ok to change the behavior here then, from "autofocus the input on page load" to "disable the component until search option data is available"?

It looks like newer development is using the latter, but still autofocusing if data is present in the cache.

Reply:
Yes, ok to change to new behavior.